query/src/storages/fuse/operations/commit.rs

```diff
@@ -33,18 +39,107 @@ use crate::storages::fuse::operations::TableOperationLog;
+        let tid = self.table_info.ident.table_id;
+
+        let mut tbl = self;
+        let mut latest: Arc<dyn Table>;
+
+        let mut retry_times = 0;
+
+        let settings = ctx.get_settings();
+
+        // The initial retry delay in millisecond. By default,  it is 5 ms.
+        let init_delay = Duration::from_millis(settings.get_storage_occ_backoff_init_delay_ms()?);
+
+        // The maximum  back off delay in millisecond, once the retry interval reaches this value, it stops increasing.
+        // By default, it is 20 seconds.
+        let max_delay = Duration::from_millis(settings.get_storage_occ_backoff_max_delay_ms()?);
+
+        // The maximum elapsed time after the occ starts, beyond which there will be no more retries.
+        // By default, it is 2 minutes
+        let max_elapsed = Duration::from_millis(settings.get_storage_occ_backoff_max_elapsed_ms()?);
+
+        // see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ for more
+        // informations. (The strategy that crate backoff implements is “Equal Jitter”)
+
+        // To simplify the settings, using fixed common values for randomization_factor and multiplier
+        let mut backoff = ExponentialBackoffBuilder::new()
+            .with_initial_interval(init_delay)
+            .with_max_interval(max_delay)
+            .with_randomization_factor(0.5)
+            .with_multiplier(2.0)
+            .with_max_elapsed_time(Some(max_elapsed))
+            .build();
+
+        loop {
+            match tbl
+                .try_commit(ctx.as_ref(), &operation_log, overwrite)
+                .await
+            {
+                Ok(_) => break Ok(()),
+                Err(e) if e.code() == ErrorCode::table_version_mismatched_code() => {
+                    match backoff.next_backoff() {
+                        Some(d) => {
+                            let name = tbl.table_info.name.clone();
+                            tracing::error!(
+                                "got error TableVersionMismatched, tx will be retried {} ms later. table name {}, identity {}",
+                                d.as_millis(),
+                                name.as_str(),
+                                tbl.table_info.ident
+                            );
+                            common_base::tokio::time::sleep(d).await;
 
-        let prev = self.read_table_snapshot(ctx.as_ref()).await?;
+                            let catalog = ctx.get_catalog();
+                            let (ident, meta) = catalog.get_table_meta_by_id(tid).await?;
+                            let table_info: TableInfo = TableInfo {
+                                ident,
+                                desc: "".to_owned(),
+                                name,
+                                meta: meta.as_ref().clone(),
+                            };
+                            latest = catalog.get_table_by_info(&table_info)?;
```

Comment:
table version is increased in `metasrv`, so here we can only use the latest table info?

Reply:
exactly